packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/mask.tests.js

```diff
@@ -1782,6 +1783,57 @@ QUnit.module('clear button', () => {
             clock.restore();
         }
     });
+
+    QUnit.test('Input text should be cleared after click on a clear button even if value is not changed yet (T1193735)', function(assert) {
+        const $textEditor = $('#texteditor').dxTextEditor({
+            mask: '43#.###',
+            showClearButton: true
+        });
+
+        const textEditor = $textEditor.dxTextEditor('instance');
+
+        const $input = $textEditor.find(`.${TEXTEDITOR_INPUT_CLASS}`);
+        const keyboard = keyboardMock($input, true);
+
+        $input.triggerHandler('focus');
+
+        caretWorkaround($input);
+        keyboard.caret(0);
+
+        keyboard.type('9');
+
+        const $clearButton = $textEditor.find(CLEAR_BUTTON_CLASS);
+
+        $clearButton.trigger('dxclick');
+
+        assert.strictEqual(textEditor.option('value'), '');
+    });
+
+    QUnit.test('Cleared text should not be restored on focusout (T1193735)', function(assert) {
```

Comment:
It's not directly related to the ticket
```suggestion
    QUnit.test('Cleared text should not be restored on focusout', function(assert) {
```